x-pack/plugins/ml/public/application/datavisualizer/index_based/components/field_data_row/action_menu/lens_utils.ts

```diff
@@ -0,0 +1,205 @@
+// eslint-disable-next-line @kbn/eslint/no-restricted-paths
+import type { IndexPatternColumn } from '../../../../../../../../lens/public/indexpattern_datasource/operations/definitions';
+// eslint-disable-next-line @kbn/eslint/no-restricted-paths
+import type { LayerConfig } from '../../../../../../../../lens/public/xy_visualization/types';
+import type { CombinedQuery } from '../../../common';
+
+interface ColumnsAndLayer {
+  columns: Record<string, IndexPatternColumn>;
+  layer: LayerConfig;
+}
+
+export function getNumberSettings(item: FieldVisConfig, defaultIndexPattern: IndexPattern) {
+  const columns: Record<string, IndexPatternColumn> = {
+    col2: {
+      dataType: 'number',
+      isBucketed: false,
+      label: `Average of ${item.fieldName}`,
+      operationType: 'avg',
+      sourceField: item.fieldName!,
+    },
+    col1: {
+      dataType: 'date',
+      isBucketed: true,
+      label: '@timestamp',
+      operationType: 'date_histogram',
+      params: { interval: 'auto' },
+      scale: 'interval',
+      sourceField: defaultIndexPattern.timeFieldName!,
+    },
+  };
+  const layer: LayerConfig = {
+    accessors: ['col2'],
+    layerId: 'layer1',
+    seriesType: 'line',
+    xAccessor: 'col1',
+  };
+
+  return { columns, layer };
+}
+export function getDateSettings(item: FieldVisConfig, defaultIndexPattern: IndexPattern) {
+  const columns: Record<string, IndexPatternColumn> = {
+    col2: {
+      dataType: 'number',
+      isBucketed: false,
+      label: 'Count of records',
+      operationType: 'count',
+      scale: 'ratio',
+      sourceField: 'Records',
+    },
+    col1: {
+      dataType: 'date',
+      isBucketed: true,
+      label: '@timestamp',
+      operationType: 'date_histogram',
+      params: { interval: 'auto' },
+      scale: 'interval',
+      sourceField: defaultIndexPattern.timeFieldName!,
+    },
+  };
+  const layer: LayerConfig = {
+    accessors: ['col2'],
+    layerId: 'layer1',
+    seriesType: 'line',
+    xAccessor: 'col1',
+  };
+
+  return { columns, layer };
+}
+
+export function getKeywordSettings(item: FieldVisConfig) {
+  const columns: Record<string, IndexPatternColumn> = {
+    col1: {
+      label: 'Top value of category',
+      dataType: 'string',
+      isBucketed: true,
+      operationType: 'terms',
+      params: {
+        orderBy: { type: 'alphabetical' },
+        size: 10,
+        orderDirection: 'asc',
+      },
+      sourceField: item.fieldName!,
+    },
+    col2: {
+      label: 'Count',
+      dataType: 'number',
+      isBucketed: false,
+      sourceField: 'Records',
+      operationType: 'count',
+    },
+  };
+  const layer: LayerConfig = {
+    accessors: ['col2'],
+    layerId: 'layer1',
+    seriesType: 'bar',
+    xAccessor: 'col1',
+  };
+
+  return { columns, layer };
+}
+
+export function getCompatibleLensDataType(type: FieldVisConfig['type']): DataType | undefined {
+  let lensType: DataType | undefined;
+  switch (type) {
```

Comment:
Is there anything we could do for `boolean` types? The top value bar chart perhaps, like you use for keyword and ip types?

Reply:
Added support for `boolean` types [here]( https://github.com/elastic/kibana/pull/89471/commits/96ff5ac3cc84ac7cc724c7cd793b40cfbe4b2b60). I default it to `true -> false` from left to right to match with the view we have now in the Data Visualizer.